### PR TITLE
Use the new function registry for the `pg_get_keywords` and `empty_row` table functions.

### PR DESCRIPTION
**Pull request by kovrus**
## Summary of the changes / Why this improves CrateDB

These table functions do not require the support for the row type signature parsing,
therefore, it is ok to migrate them even before the support for the signature parsing is implemented, see https://github.com/crate/crate/pull/9878.

## Checklist

 - [ ] Added an entry in `CHANGES.txt` for user facing changes
 - [ ] Updated documentation & `sql_features` table for user facing changes
 - [x] Touched code is covered by tests
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed
 - [x] This does not contain breaking changes, or if it does:
    - It is released within a major release
    - It is recorded in ``CHANGES.txt``
    - It was marked as deprecated in an earlier release if possible
    - You've thought about the consequences and other components are adapted
      (E.g. AdminUI)

### CODE REVIEW

File: sql/src/main/java/io/crate/expression/tablefunctions/EmptyRowTableFunction.java [hunk 1]
```diff
@@ -66,7 +61,9 @@ public FunctionInfo info() {
 
         @Override
         public RowType returnType() {
-            return emptyRow;
+            assert info.returnType() instanceof RowType
```
Comment (seut) [inline sql/src/main/java/io/crate/expression/tablefunctions/EmptyRowTableFunction.java:64]: Once https://github.com/crate/crate/pull/9853 is merged we should use the signature's return type. Probably `FunctionInfo` will be gone once all functions uses the signature registry.
Maybe we could wait a bit for #9853 as it maybe merged soon.

Comment (kovrus) [reply sql/src/main/java/io/crate/expression/tablefunctions/EmptyRowTableFunction.java:64]: 👍 Sure, i'll rebase and adjust the PRs once #9853 is merged.

Comment (seut) [reply sql/src/main/java/io/crate/expression/tablefunctions/EmptyRowTableFunction.java]: As discussed, we can't do that atm as we'd need the bound signature for the correct return type. So we have to postpone this change until we may get rid of `FunctionInfo`.

**Comment by kovrus** [comment]
@Mergifyio rebase

**Pull request merged**